### ISSUE TITLE
No server found, verify your proxy/network

### ISSUE THREAD

**Issue by bdydrp** [ddsol/speedtest.net#132]
I have had this working in node red for quite some time, and up until recently i kept getting an error:
"No server found, verify your proxy/network"
![image](https://user-images.githubusercontent.com/63593828/190157056-13e660ff-fdd1-40ce-81eb-e7548a45e69b.png)

Nothing has changed at all with my network so wondering what is causing this error?

I have tried setting a specific server ID but no go

**Comment by philly1993** [comment]
I am also seeing this as well in `1.6.2`.  This issue seemed to just appear yesterday for me.

Looks like it relates to https://github.com/ddsol/speedtest.net/blob/90fcc3599316289f6f078ae8241ecc45d9f3a01b/index.js#L649

**Comment by ryanmccartney** [comment]
Also seeing this issue on `1.6.2`.

**Comment by darvvin** [comment]
I had this running with a script every 30 minutes.  The last successful run was  Sept 16 at 4:30pm(EST).
"No server found, verify your proxy/network" ever since then.

**Comment by stefanmck** [comment]
Same problem here

**Comment by bugsounet** [comment]
I think the real question is ... why using v1.6.2 ?
because release is v2.2.0 and works fine :)

**Comment by stefanmck** [comment]
After accepting (see belwo) it works fine again

CLI options
Usage: speedtest-net [-h|--help] [--accept-license] [--server-id <id>] [--source-ip <ip>]

-h, --help: Help
--accept-license: Accept the Ookla EULA, TOS and Privacy policy. The terms only need to be accepted once.
--accept-gdpr: Accept the Ookla GDPR terms. The terms only need to be accepted once.

**Comment by philly1993** [comment]
I tried passing in those accept params in the terminal and it didn’t work for me. 

Do you have a code example of the package returning the results?

**Comment by bdydrp** [comment]
I removed the package from node red, and re-installed.
Accepted the T&C and is now working!

**Comment by VDenisyuk** [comment]
Seems that SpeedTest php endpoints do not return proper XML with config and server list anymore. They return a message, that the request limit has been exceeded from your IP. Therefore, no servers found message appears.

**Comment by KennethGrainger** [comment]
I've been going around on this problem. Lots of suggestions that almost work.
I'm running MagicMirror with Internet-Monitor, which includes speedtest-net.   
The speedtest used to work, which I presume was before Speedtest added/changed the license.
If you don't accept the license, then it doesn't return a valuable response.
I was able to get 2.2.0 kinda working, but I had to change the options in index.js to get it to work.
I had to sudo ./bin/index.js.  Then the test will run. 

If I install the global cli, it will run with --accept-license. 
But, I can't seem to get it to run with 1.6.2. It's unclear on 1.6.2, where to put the options in so they work. There are some in various places in ./index.js, but nothing I've tried has worked. The workarounds don't really resolve problems when running it as a module.

**Comment by KennethGrainger** [comment]
After going through manually updating the node_module to 2.2.0, and some other dependencies for lzma and decompress-tarxz, I was able to get this working. It'll probably get stepped on if I update it again. Might have to walk through some of the package-lock.json files if it rolls back the version.

**Comment by rbonestell** [comment]
@KennethGrainger Can you provide the libraries and versions you changed? Perhaps they can be added as overrides in our projects to get them working again. I've upgraded this dep to v2.2.0 but still get this error when running on a ras pi 4.

**Comment by adamk** [comment]
`sudo speedtest-net --accept-license`

Solved it for me

**Comment by adamk** [comment]
> Seems that SpeedTest php endpoints do not return proper XML with config and server list anymore. They return a message, that the request limit has been exceeded from your IP. Therefore, no servers found message appears.

@VDenisyuk Do you know how to fix this?

**Comment by KennethGrainger** [comment]
> @KennethGrainger Can you provide the libraries and versions you changed? Perhaps they can be added as overrides in our projects to get them working again. I've upgraded this dep to v2.2.0 but still get this error when running on a ras pi 4.

Yeah, I went through and tried to go through it all logically, step by step, trying to remove deprecation warnings and module mismatches. The who process is infuriating. I do know that I did manually run speedtest-net with sudo and --accept-license like adamk was mentioning. I have one machine running, and another that is broken. When I made it work the first time, I was just poking around all over the place getting rid of warnings. I'm not really well versed in Node. If the license is accepted, it seems that all following calls work correctly. I can't find any real explanation on what the --accept-license does, or where is stores that information.

**Comment by philly1993** [comment]
I think I have a solution - need to grab from my server but will report back with what worked for me. On Sep 24, 2023, at 11:18 AM, Adam Kimbrough ***@***.***> wrote:﻿

Seems that SpeedTest php endpoints do not return proper XML with config and server list anymore. They return a message, that the request limit has been exceeded from your IP. Therefore, no servers found message appears.

Do you know how to fix this?

**Comment by mrkarliner** [comment]
You can upload your own `servers.php` file to any web server and pass that URL to the code.  This will only use the servers you have listed.  

Example:  

```
var test = speedTest({maxTime: time, serversUrl: 'https://<server-url>/servers.php'});
```

servers.php

```
<?php 
header('Content-type: text/xml');

$server = '<settings>
<servers>
<server url="http://nycmetro-speedtest.reliablesite.net:8080/speedtest/upload.php" lat="40.5549" lon="-74.4643" name="Piscataway, NJ" country="United States" cc="US" sponsor="ReliableSite Hosting" id="32339" host="nycmetro-speedtest.reliablesite.net:8080"/>
<server url="http://stosat-plfi-01.sys.comcast.net:8080/speedtest/upload.php" lat="40.3600" lon="-74.2400" name="Plainfield, NJ" country="United States" cc="US" sponsor="Comcast" id="1769" host="stosat-plfi-01.sys.comcast.net:8080"/>
<server url="http://speedtest.isp.clairglobal.net:8080/speedtest/upload.php" lat="40.1573" lon="-76.3069" name="Lititz, PA" country="United States" cc="US" sponsor="Clair Global" id="26011" host="speedtest.isp.clairglobal.net:8080"/>
<server url="https://newyork02.speedtest.windstream.net:8080/speedtest/upload.php" lat="40.7127" lon="-74.0059" name="New York, NY" country="United States" cc="US" sponsor="Windstream" id="28032" host="newyork02.speedtest.windstream.net:8080"/>
<server url="https://speedtest.getstreamhosting.com:8080/speedtest/upload.php" lat="40.7127" lon="-74.0059" name="New York, NY" country="United States" cc="US" sponsor="ServerRoom.net" id="21313" host="speedtest.getstreamhosting.com:8080"/>
<server url="http://speedtest.w-net.us:8080/speedtest/upload.php" lat="40.7127" lon="-74.0059" name="New York, NY" country="United States" cc="US" sponsor="Wnet" id="28025" host="speedtest.w-net.us:8080"/>
<server url="https://spt-nyc.as13786.net:8080/speedtest/upload.php" lat="40.7127" lon="-74.0059" name="New York, NY" country="United States" cc="US" sponsor="Seaborn Networks" id="34335" host="spt-nyc.as13786.net:8080"/>
</servers>
</settings>';

print($server);
?>
```
